hooks/src/index.d.ts

```diff
@@ -46,8 +46,14 @@ export function useReducer<S, A, I>(
 ): [S, (action: A) => void];
 
 /** @deprecated Use the `Ref` type instead. */
-type PropRef<T> = { current: T };
-type Ref<T> = { current: T };
+type PropRef<T> = MutableRefObject<T>;
+interface Ref<T> {
+	readonly current: T | null;
+}
```

Comment:
Are we able to instead (slightly) alter `RefObject` from core?

https://github.com/preactjs/preact/blob/4aaecc18d33a9f3139ddd5f30f6d7c745076795f/src/index.d.ts#L40

If not, super nit: `RefObject` is now unused, doesn't need to be imported

https://github.com/preactjs/preact/blob/4aaecc18d33a9f3139ddd5f30f6d7c745076795f/hooks/src/index.d.ts#L1

`Ref` & `MutableRef` or `RefObject` & `MutableRefObject` would be nice for consistency's sake too, as previously mentioned.